assignments/exam2-student.html

```diff
@@ -0,0 +1,512 @@
+// example: union of [1, 2, 3, 4] and [2, 3, 4, 5] is [1, 2, 3, 4, 5]
+function union(l, r) {
+    const res = {};
+    for (let i = 0; i < l.length; i++) {
+        const lVal = l[i];
+        res[lVal] = 1;
+    }
+    for (let i = 0; i < r.length; i++) {
+        const rVal = r[i];
+        res[rVal] = 1;
+    }
+    return Object.keys(res);
+}
+</script>
+
+<script>
+// Q7 - 12 points: Fill in the missing code segments.
+//
+// this function will check user input text for equality, normalizing the input to ignore extra leading/trailing spaces and also case differences
+function compareNormalized(l, r) {
+    const lNoSpaces = l.trim();
+    const rNoSpaces = r.trim();
+    const lLowerNoSpaces = lNoSpaces.toLowerCase();
+    const rLowerNoSpaces = rNoSpaces.toLowerCase();
+    return lLowerNoSpaces === rLowerNoSpaces;
+}
+</script>
+
+<script>
+// Q8 - 24 points: Fill in the missing code segments.
+//
+// we're simulating a blackjack table - write the endRound function logic the dealer would use to find winners and pay them twice what they bet
+function cardVal(c) {
+    if (c >= 0 && c <= 9) {
+        return c;
+    } else if (c == "A") {
+        return 11;
+    } else {
+        return 10;
+    }
+}
+
+function handVal(cards) {
+    let totVal = 0;
+    for (let i = 0; i < cards.length; i++) {
+        totVal += cardVal(cards[i]);
+    }
+    if (totVal > 21) {
+        return -1;
+    }
+    return totVal;
+}
+
+function endRound(t) {
+    const dealerHandVal = handVal(t.dealer.cards); //9
+    for (let i = 0; i < t.players.length; i++) {
+        const playerHandVal = handVal(t.players[i].cards);
+        if (playerHandVal >= dealerHandVal) {
+            t.players[i].bal += t.players[i].bet * 2;
+        }
+    }
+}
+
+function simulateBlackJack() {
+    const players = [
+        {
+            "name": "Bond",
+            "cards": ["A", "J"], //win
+            "bet": 120000,
+            "bal": 10000000
+        },
+        {
+            "name": "Tank",
+            "cards": [3, 9, 2], // loose
+            "bet": 999999,
+            "bal": 1
+        },
+        {
+            "name": "Rang",
+            "cards": ["A", 2, "J", 9], // loose
+            "bet": 777,
+            "bal": 123
+        },
+    ];
+
+    const table = {
+        "players": players,
+        "dealer": {
+            "cards": ["K", 9],
+        }
+    };
+    
+    endRound(table);
+    return table;
+}
+</script>
+
+
+<script>
+// Q9 - 30 points: Program the createSizedTable function allowing the user to create a table of a specific size they specify
+// use the 'rows' and 'cols' text fields on this page to enter the desired number of rows and columns
+// build the table inside of the 'tblContainer' div
+
+function createSizedTable() {
+    const tblContainer = document.getElementById('tblContainer');
+    const rows = document.getElementById('rows').value;
```

Comment:
This is fine but also remember for the future that you can create HTML as a string and set it via `elem.innerHTML` and this is often much easier when generating content.